tests/unit/test_charm.py

```diff
@@ -29,17 +30,29 @@ def setUp(self):
         self.harness.begin()
         self.charm = self.harness.charm
 
+    @patch("charm.PostgresqlOperatorCharm._install_pip_packages")
     @patch("charm.PostgresqlOperatorCharm._install_apt_packages")
     @patch("charm.PostgresqlCluster.inhibit_default_cluster_creation")
     def test_on_install(
```

Comment:
You might want to add two tests here:

`test_on_install_pip_failure`
`test_on_install_apt_failure`

To verify that on failure they stay in the `BlockedStatus`

Reply:
Nice addition! I'll add them.